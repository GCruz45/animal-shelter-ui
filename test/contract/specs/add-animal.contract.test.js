import { provider } from "../config/init-pact";
import { Matchers } from "@pact-foundation/pact";
import { AnimalController } from "../../../controllers";
let animalBody =
  "{name: 'addedAnimal', breed: 'addedAnimalBreed', gender: 'Female', vaccinated: false,}";
describe("Animal Service", () => {
  describe("When a request to add an animal is made", () => {
    beforeAll(async () => {
      await provider.setup();
      await provider.addInteraction({
        uponReceiving: "a request to add an animal",
        state: "adds animal with name = addedAnimal",
        withRequest: {
          method: "POST",
          body: animalBody,
          path: "/animals",
        },
        willRespondWith: {
          status: 201,
          // headers: { Location: "/animals/addedAnimal" },
          body: animalBody,
        },
      });
    });
    test("should return the correct data", async () => {
      const response = await AnimalController.register(animalBody);
      expect(response.data).toMatchSnapshot();
      await provider.verify();
    });
    afterAll(() => provider.finalize());
  });
});
