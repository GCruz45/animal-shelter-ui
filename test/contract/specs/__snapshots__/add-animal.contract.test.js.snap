// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Service When a request to add an animal is made should return the correct data 1`] = `"{name: 'addedAnimal', breed: 'addedAnimalBreed', gender: 'Female', vaccinated: false,}"`;
